goal_src/levels/common/joint-exploder.gc

```diff
@@ -59,53 +63,623 @@
+             (TODO-RENAME-28 obj a1-8)
+             )
+         )
+       (TODO-RENAME-28 obj arg0)
+       )
+      )
+    )
+  (none)
+  )
+
+(defmethod TODO-RENAME-25 joint-exploder ((obj joint-exploder) (arg0 joint-exploder-list))
+  (set! (-> arg0 bbox-valid?) #f)
+  (set! (-> arg0 pre-moved?) #t)
+  (let ((s4-0 (-> obj joints))
+        (f30-0 (* (-> obj tuning gravity) (-> *display* seconds-per-frame)))
+        (s3-0 (-> arg0 head))
+        )
+    (while (>= s3-0 0)
+      (let* ((s2-0 (the-as object (+ (+ (* 176 s3-0) 12) (the-as int s4-0))))
+             (s1-0 (-> (the-as joint-exploder-joint s2-0) mat vector 3))
+             )
+        (set! (-> (the-as joint-exploder-joint s2-0) prev-pos quad) (-> s1-0 quad))
+        (+! (-> (the-as joint-exploder-joint s2-0) transv y) f30-0)
+        (vector-v+! s1-0 s1-0 (-> (the-as joint-exploder-joint s2-0) transv))
+        (let ((f0-3 0.99)
+              (f2-1 (* (-> (the-as joint-exploder-joint s2-0) rspeed) (-> *display* seconds-per-frame)))
+              (f5-0 (-> (the-as joint-exploder-joint s2-0) rmat vector 0 x))
+              (f4-0 (-> (the-as joint-exploder-joint s2-0) rmat vector 0 y))
+              (f3-0 (-> (the-as joint-exploder-joint s2-0) rmat vector 1 x))
+              (f1-2 (-> (the-as joint-exploder-joint s2-0) rmat vector 1 y))
+              )
+          (set! (-> (the-as joint-exploder-joint s2-0) rmat vector 0 x) (- (* f5-0 f0-3) (* f4-0 f2-1)))
+          (set! (-> (the-as joint-exploder-joint s2-0) rmat vector 0 y) (+ (* f5-0 f2-1) (* f4-0 f0-3)))
+          (set! (-> (the-as joint-exploder-joint s2-0) rmat vector 1 x) (- (* f3-0 f0-3) (* f1-2 f2-1)))
+          (set! (-> (the-as joint-exploder-joint s2-0) rmat vector 1 y) (+ (* f3-0 f2-1) (* f1-2 f0-3)))
+          )
+        (cond
+          ((or
+             (< (-> s1-0 y) (-> obj die-if-below-y))
+             (< (-> obj die-if-beyond-xz-dist-sqrd) (vector-vector-xz-distance s1-0 (-> obj root trans)))
+             )
+           (set! s3-0 (TODO-RENAME-24 obj arg0 s3-0))
+           )
+          (else
+            (TODO-RENAME-21 obj arg0 (the-as joint-exploder-joint s2-0))
+            (set! s3-0 (-> (the-as (pointer int16) s2-0)))
+            )
+          )
+        )
+      )
+    )
+  #f
+  )
+
+(defmethod TODO-RENAME-22 joint-exploder ((obj joint-exploder) (arg0 joint-exploder-list))
+  (dummy-13 *collide-cache* (-> arg0 bbox) (the-as uint 1) obj (the-as uint 1))
+  (let ((gp-1 (-> obj joints))
+        (v1-2 (-> arg0 head))
+        )
+    (while (>= v1-2 0)
+      (let ((s5-1 (the-as joint-exploder-joint (+ (+ (* 176 v1-2) 12) (the-as int gp-1)))))
+        (let ((s4-0 (-> s5-1 mat vector 3))
+              (a2-1 (new 'stack-no-clear 'vector))
+              (s3-0 (new 'stack-no-clear 'collide-tri-result))
+              )
+          (vector-! a2-1 s4-0 (-> s5-1 prev-pos))
+          (when (>= (dummy-18 *collide-cache* (-> s5-1 prev-pos) a2-1 40.96 1 (the-as collide-mesh-cache-tri s3-0) 1) 0.0)
```

Comment:
we should either change `s3-0` to a `collide-mesh-cache-tri` or change the signature of dummy-18 of `collide-cache` - if we mix stack types like this, we can write out of bounds on the stack, which can lead to really hard to debug corruption